File: backend/internal/models/allowance_type.go
package models

import "time"

/*
 * 〇 手当種別マスター
 *
 * 月次手当で選択できる手当種別を管理する。
 * CSV出力時の列順はDisplayOrderを使用する。
 */
type AllowanceType struct {
	ID           uint       `gorm:"primaryKey" json:"id"`
	Name         string     `gorm:"type:varchar(100);not null" json:"name"`
	DisplayOrder int        `gorm:"not null;default:0;index" json:"displayOrder"`
	IsActive     bool       `gorm:"not null;default:true;index" json:"isActive"`
	IsDeleted    bool       `gorm:"not null;default:false;index" json:"isDeleted"`
	CreatedAt    time.Time  `json:"createdAt"`
	UpdatedAt    time.Time  `json:"updatedAt"`
	DeletedAt    *time.Time `json:"deletedAt"`
}

File: frontend/src/types/admin/allowanceType.ts
/*
 * 管理者 手当種別マスター
 *
 * 月次手当登録時のドロップダウン候補を管理する。
 */

export type SearchAllowanceTypesRequest = {
  keyword: string;
  includeInactive: boolean;
  includeDeleted: boolean;
  offset: number;
  limit: number;
};

export type AllowanceTypeDetailRequest = {
  allowanceTypeId: number;
};

export type CreateAllowanceTypeRequest = {
  name: string;
  displayOrder: number;
  isActive: boolean;
};

export type UpdateAllowanceTypeRequest = {
  allowanceTypeId: number;
  name: string;
  displayOrder: number;
  isActive: boolean;
};

export type DeleteAllowanceTypeRequest = {
  allowanceTypeId: number;
};

export type AllowanceTypeResponse = {
  id: number;
  name: string;
  displayOrder: number;
  isActive: boolean;
  isDeleted: boolean;
  createdAt: string;
  updatedAt: string;
  deletedAt: string | null;
};

export type SearchAllowanceTypesResponse = {
  allowanceTypes: AllowanceTypeResponse[];
  total: number;
  offset: number;
  limit: number;
  hasMore: boolean;
};

export type AllowanceTypeDetailResponse = {
  allowanceType: AllowanceTypeResponse;
};

export type CreateAllowanceTypeResponse = {
  allowanceType: AllowanceTypeResponse;
};

export type UpdateAllowanceTypeResponse = {
  allowanceType: AllowanceTypeResponse;
};

export type DeleteAllowanceTypeResponse = {
  allowanceTypeId: number;
};
